var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -89,7 +94,6 @@ class Gcc(AutotoolsPackage):
     depends_on('gnat', when='languages=ada')
     depends_on('binutils~libiberty', when='+binutils')
     depends_on('zip', type='build', when='languages=java')
-    depends_on('zip', type='build', when='@:6 languages=all')
```

Comment:
for the record, this was a workaround for `java` introduced in https://github.com/LLNL/spack/pull/4303